xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/common/Wrappers.java

```diff
@@ -0,0 +1,169 @@
+package org.xrpl.xrpl4j.model.client.common;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.UnsignedInteger;
+import org.immutables.value.Value;
+import org.xrpl.xrpl4j.model.immutables.FluentCompareTo;
+import org.xrpl.xrpl4j.model.immutables.Wrapped;
+import org.xrpl.xrpl4j.model.immutables.Wrapper;
+
+import java.io.Serializable;
+
+@SuppressWarnings("TypeName")
+public class Wrappers {
+
+  /**
+   * This class is similar to {@link LedgerIndex} in that it represents a numerical ledger version on the XRP Ledger.
+   * However, {@link LedgerIndexBound} can be used to specify ranges of ledgers in some rippled API request objects,
+   * such as {@link org.xrpl.xrpl4j.model.client.accounts.AccountTransactionsRequestParams}. Unlike {@link LedgerIndex}
+   * which wraps an unsigned long value, this class supports {@code -1} and does not allow {@code 0}.
+   */
+  @Value.Immutable
+  @Wrapped
+  @JsonSerialize(as = LedgerIndexBound.class)
+  @JsonDeserialize(as = LedgerIndexBound.class)
+  abstract static class _LedgerIndexBound extends Wrapper<Long> implements Serializable {
```

Comment:
Should this be typed as an `UnsignedInt` instead of `Long`?

Reply:
The whole point of this class is to allow `-1` and values greater than `0`, which won't be supported by `LedgerIndex` once it just wraps an `UnsignedInteger`